wire-gradle-plugin/src/main/kotlin/com/squareup/wire/gradle/WireTask.kt

```diff
@@ -112,20 +114,27 @@ open class WireTask : SourceTask() {
         .flatMap { dep ->
           files(dep)
               .flatMap { file ->
-                if (dep !is FileCollectionDependency) {
-                  listOf(Location.get(file.path))
-                } else if (dep.files is SourceDirectorySet) {
-                  val srcDir = (dep.files as SourceDirectorySet).srcDirs.first {
-                    file.path.startsWith(it.path + "/")
-                  }
-                  listOf(Location.get(srcDir.path, file.path.substring(srcDir.path.length + 1)))
-                } else {
-                  val includes = jarToIncludes[file.path]
-                  includes?.map {
-                    Location.get(file.path, it)
-                  } ?: listOf(Location.get(file.path))
-                }
+                file.toLocations(dep)
               }
         }
   }
+
+  private fun File.toLocations(dependency: Dependency): List<Location> {
+    if (dependency is FileCollectionDependency && dependency.files is SourceDirectorySet) {
+      val srcDir = (dependency.files as SourceDirectorySet).srcDirs.first {
+        path.startsWith(it.path + "/")
+      }
+      return listOf(Location.get(
+          base = srcDir.path,
+          path = path.substring(srcDir.path.length + 1)
+      ))
+    }
+
+    val includes = dependencyToIncludes[dependency]
+        ?: return listOf(Location.get(path))
+
+    return includes.map {
+      include -> Location.get(base = path, path = include)
```

Comment:
super nit but that feels odd for the parameter to be declared here.

Reply:
I wanna leave it ’cause it’s weird that the parameter named path isn’t the path local variable.

Reply:
I meant I'm used to
```kotlin
return includes.map { include -> 
      Location.get(base = path, path = include)
}
```

Reply:
ahh, yeah that’s just a mistake.